source/common/init/manager_impl.cc

```diff
@@ -70,6 +81,23 @@ void ManagerImpl::onTargetReady() {
   }
 }
 
+void ManagerImpl::onTargetReadySendTargetName(absl::string_view target_name) {
+  // If there are no remaining targets and one mysteriously calls us back, this manager is haunted.
+  ASSERT(count_ != 0,
+         fmt::format("{} called back by target after initialization complete", target_name));
+
+  // Decrease target_name count by 1.
+  if (--target_names_count_[target_name] == 0) {
+    target_names_count_.erase(target_name);
+  }
+
+  // If there are no uninitialized targets remaining when called back by a target, that means it was
+  // the last. Signal `ready` to the handle we saved in `initialize`.
+  if (--count_ == 0) {
+    ready();
+  }
```

Comment:
just call `onTargetReady()`?

Reply:
Maybe we can just delete the original ```onTargetReady()``` since it we not be used any more?

Reply:
yep sgtm.

Reply:
Updated.